apps/arweave/src/ar_storage.erl

```diff
@@ -882,17 +883,24 @@ read_wallet_list(WalletListHash) when is_binary(WalletListHash) ->
 read_wallet_list({ok, << K:48/binary, _/binary >>, Bin}, Tree, Keys, RootHash, K) ->
 	case binary_to_term(Bin) of
 		{Key, Value} ->
-			Tree2 = ar_patricia_tree:insert(Key, Value, Tree),
+			Tree2 = ar_patricia_tree:insert(Key, Value, Tree, K),
```

Comment:
What is `K` here? It looks like we're now caching `K` in the patricia tree whereas before we weren't?

Reply:
I renamed it to NodeHash - the hash of a node in the tree. We have them on disk but we nevertheless recomputed them during node initialization instead of fetching and caching the already computed values. Note this only applies to the initial tree buildup - the hashes are cached and only the affected parts of the tree are rehashed after every update.